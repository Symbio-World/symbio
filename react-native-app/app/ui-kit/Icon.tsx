import * as React from 'react'

import { SvgProps } from 'react-native-svg'
import camera from '../../assets/svg/camera.svg'
import cross from '../../assets/svg/cross.svg'
import disk from '../../assets/svg/disk.svg'
import flashlight from '../../assets/svg/flashlight.svg'
import historyicon from '../../assets/svg/history.svg'
import fatalError from '../../assets/svg/fogg-fatal-error.svg'

type Props = SvgProps & {
  icon: Icons
}

export enum Icons {
  CAMERA = 'CAMERA',
  CROSS = 'CROSS',
  DISK = 'DISK',
  FLASHLIGHT = 'FLASHLIGHT',
  HISTORYICON = 'HISTORYICON',
  FATAL_ERROR = 'FATAL_ERROR',
}

const IconsMap = {
  [Icons.CAMERA]: camera,
  [Icons.CROSS]: cross,
  [Icons.DISK]: disk,
  [Icons.FLASHLIGHT]: flashlight,
  [Icons.HISTORYICON]: historyicon,
  [Icons.FATAL_ERROR]: fatalError,
}

export const Icon: React.FC<Props> = ({ icon, ...otherProps }) => {
  const IconComponent = IconsMap[icon]
  if (!IconComponent) throw new Error('No icon found')
  return <IconComponent {...otherProps} />
}
