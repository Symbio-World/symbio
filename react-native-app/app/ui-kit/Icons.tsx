import * as React from 'react'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

export 

