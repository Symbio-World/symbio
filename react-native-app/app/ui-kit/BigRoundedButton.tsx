import * as React from 'react'
import { TouchableOpacity, Text, View } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, Icons } from './Icon'

type Props = {
  title: string
  icon: React.ReactNode
  onPress?: () => void
  variant?: 'outline' | 'fill'
}

const ButtonStyle = {
  fill: [t.bgPrimary],
  outline: [t.bgWhite, t.border2, t.borderPrimary],
}

const TitleStyle = {
  fill: [t.textWhite],
  outline: [t.textPrimary],
}

export const BigRoundedButton: React.FC<Props> = ({
  title,
  icon,
  onPress = () => {},
  variant = 'fill',
}) => {
  const buttonStyle = ButtonStyle[variant]
  const titleStyle = TitleStyle[variant]

  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          ...buttonStyle,
          t.w24,
          t.h24,
          t.roundedFull,
          t.flexCol,
          t.itemsCenter,
          t.justifyCenter,
        ]}
      >
        <View style={[t.w6, t.h6, t.mB2]}>{icon}</View>
        <View>
          <Text style={[t.textCenter, t.textBase, t.fontBold, ...titleStyle]}>
            {title}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  )
}
