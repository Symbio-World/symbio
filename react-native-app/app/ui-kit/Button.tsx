import * as React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'

export enum ButtonStyle {
  FILL,
  OUTLINE,
  TRANSPARENT,
}

type Props = {
  onPress?: () => void
  title?: string
  icon?: IconsEnum
  variant?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const Button: React.FC<Props> = ({
  onPress = () => {},
  title = '',
  variant = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[variant]
  const innerStyle = [InnerTitleBaseStyle[variant]]
  return (
    <TouchableOpacity
      style={[
        t.p4,
        t.itemsCenter,
        t.justifyCenter,
        t.flex,
        t.flexRow,
        t.rounded,
        ...buttonStyle,
      ]}
      onPress={onPress}
    >
      {icon ? <Icon style={[t.mR2]} iconType={icon} /> : null}
      <Text style={[t.textCenter, t.text2xl, t.textWhite, ...innerStyle]}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}
