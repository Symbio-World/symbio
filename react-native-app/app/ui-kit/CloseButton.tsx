import * as React from 'react'

import { RoundedButton } from './RoundedButton'
import { ButtonStyle } from './Button'
import { Icons } from './Icon'

type CloseButtonProps = {
  onClose?: () => void
}

export const CloseButton: React.FC<CloseButtonProps> = ({ onClose }) => (
  <RoundedButton
    onPress={onClose}
    outerTitle="close"
    variant={ButtonStyle.OUTLINE}
    icon={Icons.CROSS}
  />
)
