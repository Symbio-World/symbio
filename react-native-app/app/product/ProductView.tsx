import * as React from 'react'
import { View, Image, Text, ScrollView } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { ProductData } from '@symbio/barcode-processor-core'
import { ModalActionsView } from '../ui-kit/ModalActionsView'
import { CloseButton } from '../ui-kit/CloseButton'

type Props = Omit<ProductData, 'links'> & {
  onClose?: () => void
}

export const ProductView: React.FC<Props> = ({
  image = '',
  name = '',
  description = '',
  ingredients = '',
  allergens = [],
  origin = '',
  onClose,
}) => {
  return (
    <View style={[t.flex1, t.mB3]}>
      <ScrollView style={[t.bgWhite, t.flex1]}>
        <Image
          style={[t.h56, t.bgGray200]}
          source={{ uri: image }}
          resizeMode="contain"
        />
        <View style={[t.p8]}>
          <Text style={[t.textLg, t.mB8]}>{name}</Text>
          <Text>{description}</Text>
          <Text style={[t.fontBold, t.mT8]}>Ingredients</Text>
          <Text>{ingredients ? ingredients : '-'}</Text>
          <Text style={[t.fontBold, t.mT8]}>Allergens</Text>
          {allergens.map((a) => (
            <Text>{a}</Text>
          ))}
          <Text style={[t.fontBold, t.mT8]}>Origin</Text>
          <Text>{origin}</Text>
        </View>
      </ScrollView>
      <ModalActionsView>
        <CloseButton onClose={onClose} />
      </ModalActionsView>
    </View>
  )
}
