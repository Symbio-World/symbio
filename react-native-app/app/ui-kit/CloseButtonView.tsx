import * as React from 'react'
import { t } from 'react-native-tailwindcss'
import { CloseButton } from './CloseButton'
import { View } from 'react-native'

type CloseButtonViewProps = {
  onPress?: () => void
}

export const CloseButtonView: React.FC<CloseButtonViewProps> = ({
  onPress,
}) => (
  <>
    <View style={[t.h40]} />
    <View style={[t.absolute, t.bottom0, t.wFull, t.itemsCenter, t.pB5]}>
      <CloseButton onPress={onPress} />
    </View>
  </>
)
