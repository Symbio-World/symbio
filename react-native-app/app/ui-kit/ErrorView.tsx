import * as React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { Icon, Icons } from './Icon'
import { CloseButtonView } from './CloseButtonView'

type Props = {
  error?: unknown
  onPress?: () => void
}
export const ErrorView: React.FC<Props> = ({ onPress = () => {} }) => {
  return (
    <View style={[t.flex1]}>
      <Icon icon={Icons.FATAL_ERROR} />
      <Text style={[t.textCenter, t.textBlack, t.text5xl, t.mB10]}>
        OOOOPS!
      </Text>
      <Text style={[t.textCenter, t.textBlack, t.text3xl, t.mX12]}>
        Something went wrong... You can tell us what happened or try again later
      </Text>
      <CloseButtonView onPress={onPress} />
    </View>
  )
}
