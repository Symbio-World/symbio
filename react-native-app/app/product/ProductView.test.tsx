import * as React from 'react'
import {
  render /*, fireEvent */,
  fireEvent,
} from 'react-native-testing-library'
import { ProductView } from './ProductView'
import { CloseButton } from '../ui-kit/CloseButton'
// import { Button } from '../ui-kit/Button'

describe('ProductView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ProductView />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('calls onPress', () => {
    const handleCloseButtonPress = jest.fn()
    const { getByType } = render(
      <ProductView onCloseButtonPress={handleCloseButtonPress} />,
    )
    fireEvent(getByType(CloseButton), 'onClose')
    expect(handleCloseButtonPress).toHaveBeenCalled()
  })

  //   TODO: return when onFeedbackPress used
  //   it('triggers onFeedbackPress', () => {
  //     const handleFeedbackPress = jest.fn()
  //     const { getByType } = render(
  //       <ProductView onFeedbackPress={handleFeedbackPress} />,
  //     )
  //     fireEvent.press(getByType(Button))
  //     expect(handleFeedbackPress).toHaveBeenCalledWith("Tell us what's missing")
  //   })
})
