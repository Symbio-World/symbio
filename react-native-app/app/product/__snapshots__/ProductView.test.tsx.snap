// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Array [
            Object {
              "height": 224,
            },
            Object {
              "backgroundColor": "#edf2f7",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "padding": 32,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "marginBottom": 32,
              },
            ]
          }
        >
          
        </Text>
        <Text>
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        >
          Ingredients
        </Text>
        <Text>
          -
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        >
          Allergens
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "marginTop": 32,
              },
            ]
          }
        >
          Origin
        </Text>
        <Text>
          
        </Text>
        <View
          style={
            Array [
              Object {
                "height": 160,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
